src/main/java/com/civclassic/oldenchanting/OldEnchanting.java

```diff
@@ -302,13 +306,41 @@ private void createXPBottles(Player player, int totalExp) {
 	
 	@EventHandler(priority = EventPriority.HIGHEST)
 	public void xpBottleEvent(ExpBottleEvent event) {
-		if(noExp) {
-			ProjectileSource source = event.getEntity().getShooter();
-			if(source instanceof Player) {
+		if (noExp) {
+			ThrownExpBottle bottle = event.getEntity();
+			ProjectileSource source = bottle.getShooter();
+			if (source instanceof Player) {
 				Player shooter = (Player) source;
 				shooter.giveExp(xpPerBottle);
+				bottle.teleport(shooter.getEyeLocation());
+			}
+			else if (source instanceof BlockProjectileSource) {
+				List<Player> nearby = bottle.getNearbyEntities(0.9, 0.9, 0.9)
```

Comment:
Could easily make that bigger to something like 5 blocks, ideally it'd even be a config parameter

Reply:
Done, the radius is now a config option with the recommended settings being in between two and seven